pkg/metrics/prometheus_test.go

```diff
@@ -16,212 +16,102 @@
 package metrics
 
 import (
-	"fmt"
+	"context"
 	"net/http"
-	"net/http/httputil"
-	"reflect"
+	"net/http/httptest"
 	"strings"
 	"testing"
-	"time"
-
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-func TestNewPrometheus(t *testing.T) {
-	test := struct {
-		name string
-		want Metrics
-	}{
-		name: "default",
-		want: &prom{
-			counters: make(map[string]prometheus.Counter),
-			summary:  make(map[string]prometheus.Summary),
-		},
+func TestRegisterCounter(t *testing.T) {
```

Comment:
the tests should check to ensure that the metrics are indeed set, can we include the verification of what the values should be as part of the tests as well?